helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -1165,7 +1164,11 @@ public void run() {
               + _processorName);
       while (!isInterrupted()) {
         try {
-          handleEvent(_eventBlockingQueue.take(), _cache);
+          ClusterEvent newClusterEvent = _eventBlockingQueue.take();
+          // rename the current processor's thread name to include event id
+          String currentThreadName = this.getName().split("_")[0];
+          this.setName(currentThreadName + "_" + newClusterEvent.getEventId());
+          handleEvent(newClusterEvent, _cache);
```

Comment:
Would it be better to set the event id inside handleEvent()? Logic will be more concise.
And remember that we'd better recover the thread name after the handling logic. Probably in some final block.

Reply:
Setting name before and after handle event is clearer to me. Because it is outside of handling logic and it clearly indicates the start and end position of thread renaming.